lib/src/mediator_client/mediator_client.dart

```diff
@@ -246,6 +274,35 @@ class MediatorClient {
     return subscription;
   }
 
+  /// Deletes messages from the mediator by message IDs.
+  ///
+  /// [messageHashes] - The list of message IDs to delete.
+  /// [accessToken] - Optional bearer token for authentication.
+  ///
+  /// Returns list of errors.
+  Future<List<String>> deleteMessages({
+    required List<String> messageHashes,
```

Comment:
Let's either call it `messageIDs`, if these are IDs or `messageIDHashes`.
Just as a proposal.